src/meltano/cli/run.py

```diff
@@ -85,58 +88,90 @@ async def run(
-    else:
-        raise CliError("Some ExtractLoadBlocks set failed validation.")
-    await _run_blocks(parsed_blocks, dry_run=dry_run)
+    tracker = Tracker(project)
+    cmd_ctx = cli_context_builder(
+        "run",
+        None,
+        dry_run=dry_run,
+        full_refresh=full_refresh,
+        no_state_update=no_state_update,
+        force=force,
+    )
+    with tracker.with_contexts(cmd_ctx):
+        tracker.track_command_event(cli_tracking.STARTED)
+
+        parser = BlockParser(
+            logger, project, blocks, full_refresh, no_state_update, force
+        )
+        parsed_blocks = list(parser.find_blocks(0))
+        if not parsed_blocks:
+            tracker.track_command_event(cli_tracking.ABORTED)
+            logger.info("No valid blocks found.")
+            return
+        if validate_block_sets(logger, parsed_blocks):
+            logger.debug("All ExtractLoadBlocks validated, starting execution.")
+        else:
+            tracker.track_command_event(cli_tracking.ABORTED)
+            raise CliError("Some ExtractLoadBlocks set failed validation.")
+        try:
+            await _run_blocks(tracker, parsed_blocks, dry_run=dry_run)
+        except Exception as err:
+            tracker.track_command_event(cli_tracking.FAILED)
+            raise err
+        tracker.track_command_event(cli_tracking.COMPLETED)
 
-    tracker = LegacyTracker(project)
-    tracker.track_meltano_run(blocks)
+    legacy_tracker = LegacyTracker(project)
+    legacy_tracker.track_meltano_run(blocks)
 
 
 async def _run_blocks(
-    parsed_blocks: List[Union[BlockSet, PluginCommandBlock]], dry_run: bool
+    tracker: Tracker,
+    parsed_blocks: list[BlockSet | PluginCommandBlock],
+    dry_run: bool,
 ) -> None:
     for idx, blk in enumerate(parsed_blocks):
+        blk_name = blk.__class__.__name__
+        tracking_ctx = plugins_tracking_context_from_block(blk)
+        with tracker.with_contexts(tracking_ctx):
+            tracker.track_block_event(blk_name, BlockEvents.INITIALIZED)
         if dry_run:
             if isinstance(blk, BlockSet):
                 logger.info(
                     f"Dry run, but would have run block {idx + 1}/{len(parsed_blocks)}.",
-                    block_type=blk.__class__.__name__,
+                    block_type=blk_name,
                     comprised_of=[plugin.string_id for plugin in blk.blocks],
                 )
             elif isinstance(blk, PluginCommandBlock):
                 logger.info(
                     f"Dry run, but would have run block {idx + 1}/{len(parsed_blocks)}.",
-                    block_type=blk.__class__.__name__,
+                    block_type=blk_name,
                     comprised_of=f"{blk.string_id}:{blk.command}",
                 )
             continue
 
         try:
             await blk.run()
-        except RunnerError as err:
+        except Exception as err:
             logger.error(
```

Comment:
@pnadolny13 this should catch any exceptions that come through from the lower level blocks. Previously we just captured RunnerError's only and let others bubble up.